wire/message.go

```diff
@@ -303,24 +302,30 @@ func WriteMessageN(w io.Writer, msg Message, pver uint32, dcrnet CurrencyNet) (i
 	cmd := msg.Command()
 	if len(cmd) > CommandSize {
 		msg := fmt.Sprintf("command [%s] is too long [max %v]", cmd, CommandSize)
-		return totalBytes, messageError(op, ErrCmdTooLong, msg)
+		return 0, messageError(op, ErrCmdTooLong, msg)
 	}
 	copy(elems.command[:], []byte(cmd))
 
+	// Initialize buffer with zeroed bytes for the message header (to be
+	// filled in, with checksum, after appending the payload
+	// serialization).
+	buf := new(bytes.Buffer)
+	buf.Write(make([]byte, MessageHeaderSize))
```

Comment:
I was pretty sure the make here wouldn't escape since the method takes a concrete byte slice and so it should pass escape analysis, but just to confirm:

```
./message.go:313:16: make([]byte, 24) does not escape
```

It doubt it is even worth bothering, but since it's always just writing a fixed number of zeros, it would probably technically be ever so slightly more efficient to just have a global `var zeroMessageHeader [MessageHeaderSize]byte` so that it doesn't have create a  new stack array and zero it every time and then this could just do `buf.Write(zeroMessageHeader)` .